Realm/Realm/Logging/Logger.cs

```diff
@@ -64,39 +67,73 @@ public abstract class Logger
 
+        /// <summary>
+        /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function. The message will
+        /// already be formatted with the default message formatting that includes a timestamp.
+        /// </summary>
+        /// <param name="logFunction">Function to proxy log calls to.</param>
+        /// <returns>
+        /// A <see cref="Logger"/> instance that will invoke <paramref name="logFunction"/> for each message.
+        /// </returns>
+        public static Logger Function(Action<string> logFunction) => new FunctionLogger((level, category, message) => logFunction(FormatLog(level, category, message)));
+
+        // TODO(lj): Deprecate
         /// <summary>
         /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function.
         /// </summary>
         /// <param name="logFunction">Function to proxy log calls to.</param>
         /// <returns>
         /// A <see cref="Logger"/> instance that will invoke <paramref name="logFunction"/> for each message.
         /// </returns>
-        public static Logger Function(Action<LogLevel, string> logFunction) => new FunctionLogger(logFunction);
+        public static Logger Function(Action<LogLevel, string> logFunction) => new FunctionLogger((level, _, message) => logFunction(level, message));
 
         /// <summary>
-        /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function. The message will
-        /// already be formatted with the default message formatting that includes a timestamp.
+        /// Gets a <see cref="FunctionLogger"/> that proxies Log calls to the supplied function.
         /// </summary>
         /// <param name="logFunction">Function to proxy log calls to.</param>
         /// <returns>
         /// A <see cref="Logger"/> instance that will invoke <paramref name="logFunction"/> for each message.
         /// </returns>
-        public static Logger Function(Action<string> logFunction) => new FunctionLogger((level, message) => logFunction(FormatLog(level, message)));
+        public static Logger Function(Action<LogLevel, LogCategory, string> logFunction) => new FunctionLogger(logFunction);
 
         /// <summary>
         /// Gets or sets the verbosity of log messages.
         /// </summary>
         /// <value>The log level for Realm-originating messages.</value>
         public static LogLevel LogLevel
         {
-            get => _logLevel;
+            // TODO(lj): Do we want to deprecate the getter as well?
+            get => GetLogLevel();
+            // TODO(lj): Deprecate and refer to `SetLogLevel`.
             set
             {
-                _logLevel = value;
-                SharedRealmHandle.SetLogLevel(value);
+                SetLogLevel(value);
             }
         }
 
+        public static LogCategory LogCategory
+        {
+            get => _logCategory;
+        }
```

Comment:
I don't think this makes sense

Reply:
With the removal of `_logCategory`, this field can also be removed